generator.js

```diff
@@ -59,7 +59,7 @@ module.exports = function(app, base) {
       if (err && !/Command failed/.test(err.message)) {
         cb(err);
       } else {
-        app.log.success('first commit');
+        !app.option('silent') && app.log.success('first commit');
```

Comment:
would you mind changing this to an `if` statement? I don't care if it's on one line or multiple, you can decide

Reply:
In fact I consider this more like a workaround.

I think a better approach would be to improve the `app.log` object.

But where is defined this object ? Is it from an external node module, like a base plugin ?

Let say that module exist, saying `base-log`. That object is probably an instance of a class named `BaseLog` or else, having the methods `success(), info(), error()`. Maybe each one depends on a inner generic function called with different arguments like `color, toStringFn`.

It is just some assumptions but if things work like that, I suggest the following changes:

* if `app.option('silent')` is set to `true`, then do not log anything.
* else, log as requested
* add an option argument to the function, named `force`. If that option is set to `true`, ignore the value of `app.option('silent')`

I suggest to do an other pull request on that module.

Where `app.log` is set ?

Reply:
In a such implementation, there will be zero changes made to generate-git unless upgrading the dependency for the external `app.log` object instantiation.

Reply:
Sorry I missed this reply. 

the logger is in Generate: https://github.com/generate/generate/blob/master/lib/plugins.js#L31. 

Maybe we can refactor https://github.com/base/base-logger to be much simpler use that instead?